starfish/image/_filter/linear_unmixing.py

```diff
@@ -0,0 +1,107 @@
+from functools import partial
+from typing import Optional
+
+import numpy as np
+
+from starfish.imagestack.imagestack import ImageStack
+from starfish.types import Axes
+from starfish.util import click
+from ._base import FilterAlgorithmBase
+from .util import preserve_float_range
+
+
+class LinearUnmixing(FilterAlgorithmBase):
+
+    def __init__(self, coeff_mat: np.ndarray) -> None:
+        """Image scaling filter
+
+        Parameters
+        ----------
+        coeff_mat : np.ndarray
+            matrix of the linear unmixing coefficients. Should take the form:
+            B = AX, where B are the unmixed values, A is coeff_mat and X are
+            the observed values.
+
+        """
+        self.coeff_mat = coeff_mat
+
+    _DEFAULT_TESTING_PARAMETERS = {"coeff_mat": np.array([[1, -0.25], [-0.25, 1]])}
+
+    @staticmethod
+    def _unmix(image: np.ndarray, coeff_mat: np.ndarray) -> np.ndarray:
+        """Clip values of img below and above percentiles p_min and p_max
+
+        Parameters
+        ----------
+        image : np.ndarray
```

Comment:
I think this would be great if it took an `xarray` with labeled axes. Then later you could make this change:

```diff
-        n_channels = image.shape[0]
-        for c in range(n_channels):
+        for c in image.coords[Axes.CH]:
```

Reply:
I agree that it would be nice to use the axis labels for safety, but the `apply()` passes a numpy array. I propose that we leave it as is for now and make and issue proposing that we change the `apply()` method to pass an xarray. Thoughts?

Reply:
Fixed in #1035